vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -530,38 +888,32 @@ public void setDataProvider(DataProvider<T, String> dataProvider) {
             return filterConverter.apply(filterText);
         };
 
-        // Postpone data communicator provider initialization in order to
-        // trigger item count request and data fetch upon clicking on combobox
-        dataCommunicatorInitializer = () -> {
-            dataCommunicatorInitializer = null;
-            if (lazyOpenRegistration != null) {
-                lazyOpenRegistration.remove();
-                lazyOpenRegistration = null;
+        if (lazyOpenRegistration != null) {
+            lazyOpenRegistration.remove();
+            lazyOpenRegistration = null;
```

Comment:
Why is this cleared and added again at the end of this method ?
Arguably should have not been added (again) at least when new data provider is set but not sure why it should be cleared if it not null

Reply:
Hmm, some leftover from intermediate version. I propose to add this listener only when the backend data provider is assigned and remove this registration as soon as event is triggered

Reply:
Done